### DISCUSSION TITLE
HomeSpan Remote Chicken Coop Door

### DISCUSSION THREAD

**Discussion by TheDevNino** [HomeSpan/HomeSpan#637] [Q&A]
Hello 👋

I'd like to create a remote chicken coop door. In terms of hardware, it is finished. I've acquired a 28BYJ-48 stepper motor with a driver and have developed a functional door opener using NTP. It opens in the morning and closes in the evening.

Now, I want to make the entire setup controllable with HomeKit (in addition to the automatic NTP control). This is where I came across your library. However, there's an issue: there's no WiFi connection at the chicken coop, so I need to use something like ESP Now.

Initially, I thought I could achieve this by making a few modifications to the "RemoteSensors" example. But I've realized that this approach doesn't enable the main device to send information to the remote device. Is it also possible to achieve this functionality?

**Comment by HomeSpan** [comment]
SpanPoint is fully bidirectional - you can send messages to and from any device, to and from any other device.  The Example only shows messages flowing in a single direction because that is all required for a remote temperature sensor.  But you can set both the send and receive message sizes to greater than zero, which allows for two-way communication. See the SpanPoint page for details.
